django_mongodb/query.py

```diff
@@ -332,8 +362,19 @@ def _execute_query(self):
         else:
             raise RuntimeError("Unexpected params type: %s" % params_type)
 
-        self.cursor = connection.cursor()
-        self.cursor.execute(self.sql, params)
+        # self.cursor = connection.cursor()
+        # self.cursor.execute(self.sql, params)
+
+        collection = connection.get_collection(self.model._meta.db_table)
+	
+        return collection.aggregate(self.sql)
+
+    def get_columns(self):
```

Comment:
I don't think you'll need this method. Nothing in it will work on MongoDB since we don't implement introspection. I think it's used by `RawModelIterable.__iter__()` (`self.queryset.columns`) but I think you'll need to customize the class and override `__iter__()` since the results are returned as documents rather than as tuples.

Reply:
Ah, I figure we could keep it just to have a list of the model's fields. Even if it's not being called anywhere it would still be returning an expected correct output.

On a more pertinent note, Yyu're right, it looks like the RawModelIterable can remove quite a bit from its implementation.

Reply:
```python
    def __iter__(self):
        # Cache some things for performance reasons outside the loop.
        db = self.queryset.db
        query = self.queryset.query
        connection = connections[db]
        compiler = connection.ops.compiler("SQLCompiler")(query, connection, db)
        query_iterator = iter(query)

        try:
            model_cls = self.queryset.model
            # For now, RawQuerySet should just pull its columns
            fields = [self.queryset.model_fields.get(c) for c in self.queryset.columns]
            # Not sure if converters are useful. if I understand correctly this is how
            # we get from db.column_name --> models.Model.field_name conversion         
            converters = compiler.get_converters(
                [f.get_col(f.model._meta.db_table) if f else None for f in fields]
            )
            if converters:
                query_iterator = compiler.apply_converters(query_iterator, converters)
            for values in query_iterator:
                # Associate fields to values
                model_init_values = [values[pos] for pos in model_init_pos]
                instance = model_cls.from_db(db, model_init_names, model_init_values)
                if annotation_fields:
                    for column, pos in annotation_fields:
                        setattr(instance, column, values[pos])
                yield instance
        finally:
            # Done iterating the Query. If it has its own cursor, close it.
            if hasattr(query, "cursor") and query.cursor:
                query.cursor.close()
```

Reply:
Cautiously declaring done in 832a6a9eb189446cd64f956fd5fad70d6828d4e9